src/WPGraphQL.php

```diff
@@ -640,33 +622,45 @@ public static function get_allowed_post_types( $output = 'names', $args = [] ) {
 			$allowed_post_type_names = apply_filters( 'graphql_post_entities_allowed_post_types', $post_type_names, $post_type_objects );
 
 			// Filter the post type objects if the list of allowed types have changed.
-			if ( $post_type_names !== $allowed_post_type_names ) {
-				// Maybe they're just out of order.
-				sort( $post_type_names );
-				sort( $allowed_post_type_names );
-
-				if ( $post_type_names !== $allowed_post_type_names ) {
-					$post_type_objects = array_filter($post_type_objects, function ( $obj ) use ( $allowed_post_type_names ) {
-						return in_array( $obj->name, $allowed_post_type_names, true );
-					});
+			$post_type_objects = array_filter( $post_type_objects, static function ( $obj ) use ( $allowed_post_type_names ) {
+
+				if ( empty( $obj->graphql_plural_name ) && ! empty( $obj->graphql_single_name ) ) {
+					$obj->graphql_plural_name = $obj->graphql_single_name;
```

Comment:
Can we drop a comment here pointing to where it gets renamed to `all{graphql_single_name}` to improve readability?

Reply:
I'm worried that setting it there and not transforming it is going to cause issues with people using `$obj->graphql_plural_name` in their own code, since its no longer the source of truth 🤔